contracts/ceros/provider/SlisBNBLpProvider.sol

```diff
@@ -0,0 +1,262 @@
+     */
+    function isUserLpSynced(address _account) external view returns (bool) {
+        uint256 totalTokenBalance = dao.locked(token, _account);
+        uint256 userTotalLp = totalTokenBalance * exchangeRate / RATE_DENOMINATOR;
+        uint256 expectUserPart = userTotalLp * userLpRate / RATE_DENOMINATOR;
+        uint256 expectReservePart = userTotalLp - expectUserPart;
+
+        return userLp[_account] == expectUserPart && userReservedLp[_account] == expectReservePart;
+    }
+
+    /**
+     * @dev considering burn and mint with delta requires more if-else branches
+     * so here just burn all and re-mint all
+     *
+     * @param _account user address to sync
+     */
+    function _syncLp(address _account) internal override returns (bool) {
+        uint256 totalTokenBalance = dao.locked(token, _account);
+        uint256 userTotalLp = totalTokenBalance * exchangeRate / RATE_DENOMINATOR;
+        uint256 expectUserLp = userTotalLp * userLpRate / RATE_DENOMINATOR;
+        uint256 expectReserveLp = userTotalLp - expectUserLp;
+        uint256 reservePart = userReservedLp[_account];
+        uint256 userPart = userLp[_account];
+        if (userPart == expectUserLp && reservePart == expectReserveLp) {
+            return false;
+        }
+
+        // burn all first
+        if (reservePart > 0) {
+            lpToken.burn(lpReserveAddress, reservePart);
+            userReservedLp[_account] = 0;
+            totalReservedLp -= reservePart;
+        }
+
+        Delegation storage delegation = delegation[_account];
+        uint256 currentDelegateLp = delegation.amount;
+        address currentDelegateTo = delegation.delegateTo;
+        if (userPart > 0) {
+            if (currentDelegateLp > 0) {
+                lpToken.burn(currentDelegateTo, currentDelegateLp);
+                delegation.amount = 0;
+            }
+            uint256 userSelf = userPart - currentDelegateLp;
+            if (userSelf > 0) {
+                lpToken.burn(_account, userSelf);
+            }
+            userLp[_account] = 0;
+        }
+
+        // re-mint
+        if (expectReserveLp > 0) {
+            lpToken.mint(lpReserveAddress, expectReserveLp);
+            userReservedLp[_account] = expectReserveLp;
+            totalReservedLp += expectReserveLp;
+        }
+
+        uint256 expectDelegateLp = userPart > 0 ? expectUserLp * currentDelegateLp / userPart : 0;
+        uint256 expectUserSelfLp = expectUserLp - expectDelegateLp;
+        if (expectDelegateLp > 0) {
+            lpToken.mint(currentDelegateTo, expectDelegateLp);
+            delegation.amount = expectDelegateLp;
+        }
+        if (expectUserSelfLp > 0) {
+            lpToken.mint(_account, expectUserSelfLp);
+        }
+
+        userLp[_account] = expectUserLp;
+        emit SyncUserLpWithReserve(_account, expectUserLp, expectReserveLp);
+        return true;
+    }
+
+    function changeExchangeRate(uint128 _exchangeRate) external onlyRole(DEFAULT_ADMIN_ROLE) {
```

Comment:
the role of changeExchangeRate,changeUserLpRate and changeLpReserveAddress should be `manager`